### PR TITLE
Add audit log for alerts

### PR DESCRIPTION
**Pull request by eric3873**

### CODE REVIEW

**Review by perrettecl — APPROVED** [review]

File: src/fty-alert-engine.cfg.in [hunk 1]
```diff
@@ -6,4 +6,4 @@ server
     workdir = .         #   Working directory for daemon
     verbose = 0         #   Do verbose logging of activity?
 log
-    config = "/etc/fty/ftylog.cfg"     #   Path to the log configuration file (optional)
\ No newline at end of file
+    config = "/etc/fty/fty-alert-engine/alertsenginelog.cfg"     #   Path to the log configuration file (optional)
```
Comment (jeffdegott) [inline src/fty-alert-engine.cfg.in:9]: shall be named fty-alert-engine-log.cfg

File: include/fty_alert_engine_audit_log.h [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+/* Prints message in Audit Log with FATAL level. */
+#define log_fatal_alarms_engine_audit(...) \
+        log_fatal_log(AlertsEngineAuditLogManager::getInstance(), __VA_ARGS__);
+
+//singleton for logger management
+class AlertsEngineAuditLogManager
+{
+private:
+    AlertsEngineAuditLogManager () = default;
+    ~AlertsEngineAuditLogManager () = default;
```
Comment (jeffdegott) [inline include/fty_alert_engine_audit_log.h:57]: set _alertsauditlog to NULL in constructor  or do some _alertsauditlog init in declaration

File: src/fty_alert_engine_audit_log.cc [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+*/
+
+#include "fty_alert_engine_audit_log.h"
+
+Ftylog *AlertsEngineAuditLogManager::_alertsauditlog = nullptr;
+
+//  init audit logger
+void AlertsEngineAuditLogManager::init (const char* configLogFile)
+{
+    _alertsauditlog = ftylog_new ("alerts-engine-audit", configLogFile);
```
Comment (jeffdegott) [inline src/fty_alert_engine_audit_log.cc:36]: protect new call with   if (!_alertsauditlog)

File: src/luarule.cc [hunk 4]
```diff
@@ -87,45 +88,59 @@ void LuaRule::code (const std::string &newCode)
 int LuaRule::evaluate (const MetricList &metricList, PureAlert &pureAlert)
 {
     log_debug("LuaRule::evaluate %s", _name.c_str());
+    int res = 0;
```
Comment (jeffdegott) [inline src/luarule.cc:91]: here we assume RULE_RESULT_UNKNOWN != 0

**Review by jeffdegott — APPROVED** [review]

File: src/fty-alert-engine.cfg.in (hunk 1 above)
Comment (eric3873) [reply src/fty-alert-engine.cfg.in:9]: Take into account

File: src/fty_alert_engine_audit_log.cc (hunk 3 above)
Comment (eric3873) [reply src/fty_alert_engine_audit_log.cc:36]: Take into account

File: src/luarule.cc (hunk 4 above)
Comment (eric3873) [reply src/luarule.cc:91]: No definition for result good (=0), should be added

**Pull request merged**